client/app/scripts/liveblog-advertising/controllers/advertising.js

```diff
@@ -96,18 +108,128 @@ upload, $templateCache, freetypeService, modal) {
         if ($scope.advert._id) {
             // we are editing existing ad
             api('advertisements').save($scope.advert, newAd).then(function(data) {
-                handleSaveSuccess();
+                handleAdvertSaveSuccess();
             }, function(data) {
-                handleSaveError();
+                handleAdvertSaveError();
             });
         } else {
             api('advertisements').save(newAd).then(function(data) {
-                handleSaveSuccess();
+                handleAdvertSaveSuccess();
             }, function(data) {
-                handleSaveError();
+                handleAdvertSaveError();
             });
         }
     }
 
     loadAdverts();
+
+
+    //COLLECTIONS
+    function loadCollections() {
+        $scope.collectionsLoading = true;
+        api('collections').query({where: {deleted: false}}).then(function(data) {
+            $scope.collections = data._items;
+            notify.info('Collections loaded');
+            $scope.collectionsLoading = false;
+        }, function(data) {
+            $scope.collectionsLoading = false;
+            notify.error(gettext('There was an error getting the adverts'));
+        })
+    }
+
+    function handleCollectionSaveSuccess() {
+        notify.info(gettext('Collection saved successfully'));
+        $scope.collection = {};
+        $scope.collectionModalActive = false;
+        $scope.dialogCollectionLoading = false;
+        loadCollections();
+    }
+
+    function handleCollectionSaveError() {
+        notify.error(gettext('Something went wrong, please try again later!'), 5000)
+    }
+
+    $scope.openCollectionDialog = function(collection) {
+        // load all available adverts without showing any messages
+        loadAdverts(true);
```

Comment:
Shouldn't you be waiting for the adverts to be loaded before, executing the next instructions?